ravenframework/Optimizers/BayesianOptimizer.py

```diff
@@ -0,0 +1,758 @@
+    toAdd['solutionDeviation'] = self._optValSigma
+    toAdd['modelRuns'] = self._evaluationCount
+    return toAdd
+
+  def incrementIteration(self, traj):
+    """
+      Increments the "generation" or "iteration" of an optimization algorithm.
+      The definition of generation is algorithm-specific; this is a utility for tracking only.
+      @ In, traj, int, identifier for trajectory
+      @ Out, None
+    """
+    self._iteration[traj] += 1
+
+  def getIteration(self, traj):
+    """
+      Provides the "generation" or "iteration" of an optimization algorithm.
+      The definition of generation is algorithm-specific; this is a utility for tracking only.
+      @ In, traj, int, identifier for trajectory
+      @ Out, counter, int, iteration of the trajectory
+    """
+    return self._iteration[traj]
+
+  #################################
+  # Model Training and Evaluation #
+  #################################
+  def _setModelBounds(self):
+    """
+      Corrects hyperparameter bounds to account for normalized input space
+      @ In, None
+      @ Out, None
+    """
+    # NOTE This assumes scikitlearn GPR model
+    hyperParamList = self._model.supervisedContainer[0].model.kernel.hyperparameters
+    for hyperParam in hyperParamList:
+      if 'length_scale' in hyperParam.name:
+        hyperBound = hyperParam.name + '_bounds'
+        self._model.supervisedContainer[0].model.kernel.set_params(**{hyperBound:(1e-5,1)})
+
+  def _logMarginalLikelihood(self, logTheta):
+    """
+      Evaluates the log marginal likelihood of the sklearn GPR model
+      @ In, logTheta, np.array, log transformed hyperparameters
+      @ Out, logLikelihood, float, log-marginal likelihood
+      @ Out, lmlGradient, gradient of log-marginal likelihood
+    """
+    result = self._model.supervisedContainer[0].model.log_marginal_likelihood(logTheta, eval_gradient=True, clone_kernel=False)
+    # Flipping sign for maximization
+    logLikelihood = -1 * result[0]
+    lmlGradient = -1 * result[1]
+    return logLikelihood, lmlGradient
+
+  def _selectHyperparameters(self, restartCount=5):
+    """
+      Selects MAP model in model space, and is exclusively for sklearn GPR models
+      @ In, restartCount, int, number of initial points to start optimization from
+      @ Out, None
+    """
+    # Build bounds for optimization
+    paramBounds = []
+
+    for hyperParam in self._model.supervisedContainer[0].model.kernel.hyperparameters:
+      for bound in hyperParam.bounds:
+        paramBounds.append(tuple(np.log(bound)))
+
+    # Restart locations include current parameter values
+    sampler = LHS(xlimits=np.array(paramBounds), criterion='cm', random_state=self._seed)
+    initSamples = sampler(restartCount-1)
+    currentTheta = np.array([self._model.supervisedContainer[0].model.kernel.theta])
+    initSamples = np.concatenate((initSamples, currentTheta))
+    # Selecting MAP for each restart
+    options = {'ftol':1e-10, 'maxiter':200, 'disp':False}
```

Comment:
I would suggest put this line to __init__ method, so we can keep tracking the settings in one place.